src/Host/Broker/Impl/Interpreters/InterpretersController.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.R.Core.Install;
+using Microsoft.Extensions.Options;
+using Microsoft.R.Host.Broker.Security;
+
+namespace Microsoft.R.Host.Broker.Interpreters {
+    [Authorize(Policy = Policies.RUser)]
+    [Route("/interpreters")]
+    public class InterpretersController : Controller {
+        private readonly IOptions<InterpretersOptions> _interpOptions;
+        private readonly RInstallation rInstallation = new RInstallation();
+
+        public InterpretersController(IOptions<InterpretersOptions> interpOptions) {
+            _interpOptions = interpOptions;
+        }
+
+        [HttpGet]
+        public IEnumerable<InterpreterInfo> Get() {
+            if (_interpOptions.Value.AutoDetect) {
```

Comment:
if AutoDetect, we may potentially return the same interpreter twice?  Is that okay?